### PR TITLE
feat(Snowflake): Added feature flags to toggle snowflake ids

### PR DESCRIPTION
**Pull request by maxiuyuan**
Added feature flags to toggle snowflake ids. This will allow us to test snowflake ids on our own internal test orgs

### CODE REVIEW

**Review by leedongwei — APPROVED** [review]

File: src/sentry/models/organization.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def save(self, *args, **kwargs):
                 slugify_target = slugify_target.replace("_", "-").strip("-")
                 slugify_instance(self, slugify_target, reserved=RESERVED_ORGANIZATION_SLUGS)
 
-        if SENTRY_USE_SNOWFLAKE:
+        if SENTRY_USE_SNOWFLAKE or features.has("organizations:team-roles", self):
```
Comment (dashed) [inline src/sentry/models/organization.py:203]:
```suggestion
        if SENTRY_USE_SNOWFLAKE or features.has("organizations:enable-snowflake-id", self):
```

File: src/sentry/models/project.py [hunk 2]
```diff
@@ -180,7 +180,7 @@ def save(self, *args, **kwargs):
                     max_length=50,
                 )
 
-        if SENTRY_USE_SNOWFLAKE:
+        if SENTRY_USE_SNOWFLAKE or features.has("organizations:team-roles", self.organization):
```
Comment (dashed) [inline src/sentry/models/project.py:183]:
```suggestion
        if SENTRY_USE_SNOWFLAKE or features.has("organizations:enable-snowflake-id", self.organization):
```

**Comment by dashed** [comment]
@maxiuyuan What's the difference between `SENTRY_USE_SNOWFLAKE` and `organizations:enable-snowflake-id`?

**Comment by maxiuyuan** [comment]
> @maxiuyuan What's the difference between `SENTRY_USE_SNOWFLAKE` and `organizations:enable-snowflake-id`

We probably don't need the first one anymore since the feature flag could control it but that was a flag that we could toggle in settings for everyone, where as the feature flag is to do controlled rollouts

**Review by dashed — APPROVED** [review]

**Review by RyanSkonnord — APPROVED** [review]

**Pull request merged**